src/coreclr/inc/gcinfodecoder.h

```diff
@@ -301,10 +303,19 @@ class BitStreamReader
 
         _ASSERTE(numBits > 0 && numBits <= BITS_PER_SIZE_T);
 
+        // "m_RelPos == BITS_PER_SIZE_T" means that m_current
```

Comment:
This is not strictly needed. `newRelPos > BITS_PER_SIZE_T` path below would take care of it too.

Is having an extra (potentially mispredicted) branch here worth it?

Reply:
It just did not occur to me that `newRelPos > BITS_PER_SIZE_T` is safe to do even if we need to prefetch. Removing this branch makes us computationally similar to the previous solution. Except `Skip` is more complex, but `Skip` is rare.   
This makes me completely happy with the change.  Thanks!